src/components/Card/Card.tsx

```diff
@@ -11,116 +11,225 @@ export interface MyFormValues {
+  const initialValues: MyFormValues = {
+    cardNumber: "",
+    cardHolder: "",
+    CVV: "",
+    month: "",
+    year: "",
+  };
 
   return (
     <div>
-        <h2 className={isDarkModeActive ? "header light-text" : "header"}>Visa Card</h2>
-        
-        <div className={isDarkModeActive ? "card-validator light-border" : "card-validator"}>
-          <Formik
-            initialValues={initialValues}
-            validationSchema={Yup.object({
-              cardNumber: Yup.string()
-                .label('Card Number')
-                .max(16)
-                .min(16)
-                .matches(/^[0-9]*$/, 'Only digits')
-                .required('Required'),
-              cardHolder: Yup.string()
-                .label('Card Holder')
-                .max(20, 'Must be 20 characters or less')
-                .min(2, 'Must be 2 characters or more')
-                .matches(/^[aA-zZ\s]+$/, 'Wrong format')
-                .required('Required'),
-              month: Yup.string()
-                .label('Expiration Date')
-                .required('Required'),
-              year: Yup.string()
-                .label('Expiration Date')
-                .required('Required'),
-              CVV: Yup.string()
-                .label('CVV Number')
-                .min(3, 'Must be 3 or 4 digits')
-                .max(4, 'Must be 3 or 4 digits')
-                .required('Required'),
-            })}
-            onSubmit={(values, {setSubmitting}) => {
-              setTimeout(() => {
-                alert(JSON.stringify(values, null, 2));
-                setSubmitting(false);
-              }, 500)
-            }}
-          >
-            {({ values, isSubmitting }) => (
-              <Form className='card-fields'>
-                <CardView values={values} isDarkModeActive={isDarkModeActive} />
-                <label className={isDarkModeActive ? 'card-field-label light-text' : 'card-field-label dark-text'} htmlFor="cardNumber">Card Number</label>
-                <Field className='card-field' name="cardNumber" type="text" />
-                <span className={isDarkModeActive ? 'err-msg light-err' : 'err-msg'}><ErrorMessage name="cardNumber" /></span>
-              
-                <label className={isDarkModeActive ? 'card-field-label light-text' : 'card-field-label dark-text'} htmlFor="cardHolder">Card Holder</label>
-                <Field className='card-field' name="cardHolder" type="text" />
-                <span className={isDarkModeActive ? 'err-msg light-err' : 'err-msg'}><ErrorMessage name="cardHolder" /></span>
+      <h2
+        className={isDarkModeActive ? "header light-text" : "header"}
+        data-testid="header"
+      >
+        Visa Card
+      </h2>
+
+      <div
+        className={
+          isDarkModeActive ? "card-validator light-border" : "card-validator"
+        }
+      >
+        <Formik
+          initialValues={initialValues}
+          validationSchema={Yup.object({
```

Comment:
you can move out validation function out of component, the component function should contain only necessary parts, the best practice is to not define anything inside the component itself